core/storage/statistics/gae_models.py

```diff
@@ -1679,19 +1794,20 @@ def get_entity_id(cls, exploration_id, exploration_version):
     def create(
             cls, exp_id, version, num_starts, num_completions,
             state_hit_counts):
+        # type: (Text, Text, int, int, Dict[Text, int]) -> None
         """Creates a new ExplorationAnnotationsModel and
         then writes it to the datastore.
 
         Args:
             exp_id: str. ID of the exploration currently being played.
-            version: int. Version of exploration.
+            version: str. Version of exploration.
             num_starts: int. Number of students who started the exploration.
             num_completions: int. Number of students who have completed
                 the exploration.
             state_hit_counts: dict. Describes the number of hits
                 for each state.
         """
-        entity_id = cls.get_entity_id(exp_id, version)
+        entity_id = cls.get_entity_id(exp_id, int(version))
```

Comment:
That's what I thought at first. But the storage model stores it as string. See [here](https://github.com/oppia/oppia/blob/f10087ee6939e867a4381e3a767446ea39281234/core/storage/statistics/gae_models.py#L1648).

Reply:
But my point (as I stated earlier) is that you are still typecasting it to integer (and it's working fine). Even the method earlier defined `int` as its type in docstring. Had a look at [tests](https://github.com/oppia/oppia/blob/7bd323854f82f1ebb435a7a5c14838f636da906b/core/storage/statistics/gae_models_test.py#L530) too and even they provide strs of integers as arguments. Have you tried modifying the attribute type to IntegerProperty? Tests would have to be modified too then.

(cc @seanlip - as the code owner please see if this is going in the right direction, tests might be misleading here, so not sure. My suggestions are only based on observations.)

Reply:
Regarding changing of datatype in storage model, I don't think it is allowed generally? 
And if we want to change this, we will have to:
1. Change all the occurences of this function.
2. Write a one off job to change all the entities in the model

@seanlip Please let me know if I'm wrong. Thanks!

Reply:
@hardikkat24 is correct that we can't change the data type in the storage model. We'll just have to live with the inconsistency for now and I think the cast that you have is probably the best thing to do given the current situation.

I have a feeling that this model might get updated anyway though since we're moving to Py3 and Beam. Could you please add a TODO above the "version" field and file an issue saying something like "TODO(...): Determine whether this can be updated to use an int version instead, and update existing datastore models if needed." That way, when someone looks at this in the future and thinks about the stats pipeline, this issue can be flagged. Thanks!

Reply:
Adding an extra comment here so that this thread also shows up as part of my more comprehensive review pass (to help with bookkeeping).

Reply:
Done!